lib/msf/core/exploit/format/php_payload_png.rb

```diff
@@ -0,0 +1,101 @@
+# -*- coding: binary -*-
+
+require 'chunky_png'
+
+module Msf::Exploit::Format::PhpPayloadPng
+  # This mixin module provides methods to inject persistent PHP payloads into a PNG file.
+  # It is based on the article of Quentin Roland from SynActiv.
+  # https://www.synacktiv.com/en/publications/persistent-php-payloads-in-pngs-how-to-inject-php-code-in-an-image-and-keep-it-there.html
+  # The mixin depends on the GEM libary ChunkyPNG that provides the basic PNG image processing functionality.
+  #
+  # There are five methods of code injection described in the article:
+  # 1: Inject PHP payload into the PNG comment field
+  # 2: Inject PHP payload at the end of the PNG file, the so called raw insertion
+  # 3: Inject PHP payload in the PLTE chunk of the PNG file
+  # 4: Inject PHP payload in the IDAT chunk of the PNG file
+  # 5: Inject PHP payload in a random tEXT chunk of the PNG file
+  #
+  # Method 1 and 2 will not survive any image compression configured and applied by a PHP web application
+  # Method 3 will survive image compression, but no image resizing configured and applied by a PHP web application
+  # Method 4 will survive all compression and resizing but payload is fixed and restricted.
+  # Method 5 will survive Imagick resizing
+
+  # In the module below, we will offer only three (3) methods e.g, Raw, PLTE and tEXt for which we will combine method 1 and 5
+  # TODO: IDAT chunk payload injection has most potential but is not flexible and is fixed for payloads that can be injected.
+  #
+  #                     No processing   PHP-GD compression      PHP-GD resizing         Imagick resizing
+  # Raw insertion        ✅                    ❌                  ❌                     ❌
+  # PLTE chunk           ✅                    ✅                  ❌                     ❌
+  # TODO: IDAT chunk     ✅                    ✅                  ✅                     ✅
+  # tEXt chunk           ✅                    ❌                  ❌                     ✅
+  #
+  # inject_php_payload_png(payload, injection_method)
+  # @param payload [String] to inserted
+  # @param injection_method [String] accepting only standard values 'RAW', 'PLTE', or 'TEXT'
+  # @return [String, nil] PNG binary string if injection is succesfull, otherwise return nil with an error message
+
+  def inject_php_payload_png(payload, injection_method)
```

Comment:
Making `injection_method` a keyword argument with a default value may be more user-friendly.

The `IDAT` technique, once implemented, would be the best default value as the most reliable technique.